KruFPS/KruFPS.cs

```diff
@@ -211,36 +213,64 @@ public override void ModSettings()
+            //So we can do stuff Before SaveGame event is fired,
+            //In PrepareForSaveGame() function.
+            FsmHook.FsmInject(GameObject.Find("REPAIRSHOP").transform.Find("LOD/SHITHOUSE/SavePivot").GetChild(0).gameObject, "Mute audio", PrepareForSaveGame);
+            if (!GameObject.Find("STORE").transform.Find("LOD/SHITHOUSE/SavePivot").GetChild(0).gameObject.activeSelf)
+                GameObject.Find("STORE").transform.Find("LOD/SHITHOUSE/SavePivot").GetChild(0).gameObject.SetActive(true);
+            FsmHook.FsmInject(GameObject.Find("STORE").transform.Find("LOD/SHITHOUSE/SavePivot").GetChild(0).gameObject, "Mute audio", PrepareForSaveGame);
+            FsmHook.FsmInject(GameObject.Find("COTTAGE").transform.Find("SHITHOUSE/SavePivot").GetChild(0).gameObject, "Mute audio", PrepareForSaveGame);
+            if (!GameObject.Find("CABIN").transform.Find("SHITHOUSE/SavePivot").GetChild(0).gameObject.activeSelf)
+                GameObject.Find("CABIN").transform.Find("SHITHOUSE/SavePivot").GetChild(0).gameObject.SetActive(true);
+            FsmHook.FsmInject(GameObject.Find("CABIN").transform.Find("SHITHOUSE/SavePivot").GetChild(0).gameObject, "Mute audio", PrepareForSaveGame);
+            FsmHook.FsmInject(GameObject.Find("LANDFILL").transform.Find("SHITHOUSE/SavePivot").GetChild(0).gameObject, "Mute audio", PrepareForSaveGame);
+            FsmHook.FsmInject(GameObject.Find("YARD").transform.Find("Building/LIVINGROOM/LOD_livingroom/SAVEGAME").gameObject, "Mute audio", PrepareForSaveGame);
+            if(GameObject.Find("JAIL") != null)
+                FsmHook.FsmInject(GameObject.Find("JAIL/SAVEGAME"), "Mute audio", PrepareForSaveGame);
 
-        public override void OnSave()
+        }
+        private void PrepareForSaveGame()
         {
-            // Called once, when save and quit
-            // Serialize your save file here.
-            foreach (var item in awayFromHouse)
+            Debug.Log("[KruFPS] Prepare for save started!");
+            //Prepare everything here for SAVEGAME event
+            try
             {
-                EnableDisable(item, true); //ENABLE
-            }
-            foreach (var item in gameObjects)
-            {
-                EnableDisable(item, true); //ENABLE
+                foreach (GameObject item in awayFromHouse)
+                {
+                    EnableDisable(item, true); //ENABLE
+                }
+                foreach (GameObject item in gameObjects)
+                {
+                    EnableDisable(item, true); //ENABLE
+                }
+                FERNDALE.EnableDisable(true);
+                FLATBED.EnableDisable(true);
+                GIFU.EnableDisable(true);
+                HAYOSIKO.EnableDisable(true);
+                JONNEZ.EnableDisable(true);
+                KEKMET.EnableDisable(true);
+                RUSKO.EnableDisable(true);
+                CABIN.SetActive(true);
+                KINEMATIC.isKinematic = true;
+
```

Comment:
Axles and Car Dynamics (See https://github.com/Krutonium/KruFPS/blob/master/KruFPS/KruFPS.cs#L40 )